### PR TITLE
chore: Reduce dependencies and binary size, add circle ci detector

### PR DESCRIPTION
**Pull request by mjhenkes**
- Closes 

### Additional details

This cleans up our open telemetry dependencies (which i should have done before the last pr) it also reduces the size of the open telemetry modules included in the binary from 9.7 mb to 2.4 mb

### How has the user experience changed?

n/a

### PR Tasks

- [ x ] Have tests been added/updated?
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [ ] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

File: .circleci/workflows.yml [hunk 1]
```diff
@@ -493,7 +493,6 @@ commands:
               # internal PR
               CYPRESS_RECORD_KEY=$MAIN_RECORD_KEY \
               CYPRESS_INTERNAL_ENABLE_TELEMETRY="true" \
-              OTEL_RESOURCE_ATTRIBUTES="ci.branch=$CIRCLE_BRANCH,ci.job=$CIRCLE_JOB,ci.node-index=$CIRCLE_NODE_INDEX,ci.circle=$CIRCLECI,ci.build-url=$CIRCLE_BUILD_URL,ci.build-number=$CIRCLE_BUILD_NUM" \
```
Comment (mjhenkes) [inline .circleci/workflows.yml:496]: the circle ci detector replaces the need for this line

File: packages/telemetry/package.json [hunk 2]
```diff
@@ -16,13 +16,14 @@
   },
   "dependencies": {
     "@opentelemetry/api": "1.4.1",
-    "@opentelemetry/auto-instrumentations-node": "0.36.4",
```
Comment (mjhenkes) [inline packages/telemetry/package.json:19]: specifically we don't need this dep

File: packages/telemetry/package.json [hunk 3]
```diff
@@ -16,13 +16,14 @@
...
-    "@opentelemetry/exporter-trace-otlp-http": "0.36.1",
-    "@opentelemetry/instrumentation": "0.36.1",
```
Comment (mjhenkes) [inline packages/telemetry/package.json:21]: or this one

File: scripts/binary/binary-cleanup.js [hunk 4]
```diff
@@ -201,6 +201,8 @@ const buildEntryPointAndCleanup = async (buildAppDir) => {
     path.join(buildAppDir, '**', 'nexus', 'dist-esm'),
     path.join(buildAppDir, '**', '@graphql-tools', '**', '*.mjs'),
     path.join(buildAppDir, '**', 'graphql', '**', '*.mjs'),
+    path.join(buildAppDir, '**', '@openTelemetry', '**', 'esm'),
+    path.join(buildAppDir, '**', '@openTelemetry', '**', 'esnext'),
```
Comment (mjhenkes) [inline scripts/binary/binary-cleanup.js:205]: these lines remove the esm and esnext folders from open telemetry, we don't use them.

**Review by emilyrohrbough — APPROVED** [review]

File: packages/telemetry/test/detectors/circleCiDetectorSync.spec.ts [hunk 5]
```diff
@@ -0,0 +1,108 @@
+import { expect } from 'chai'
+
+process.env.CIRCLECI
```
Comment (AtofStryker) [inline packages/telemetry/test/detectors/circleCiDetectorSync.spec.ts]: Is this doing anything?
```suggestion
```

File: packages/telemetry/test/detectors/circleCiDetectorSync.spec.ts [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+      delete process.env.CIRCLE_NODE_INDEX
+      delete process.env.CIRCLE_BUILD_URL
+      delete process.env.CIRCLE_BUILD_NUM
+      delete process.env.CIRCLE_SHA1
+      delete process.env.CIRCLE_PR_NUMBER
+    })
+
+    afterEach(() => {
+      // Replace values
+      process.env.CIRCLECI = processValues.CIRCLECI
```
Comment (AtofStryker) [inline packages/telemetry/test/detectors/circleCiDetectorSync.spec.ts:34]: Does this condition need to be reversed so we delete the circle process variables so they don't leak out to other tests in the process?

Comment (mjhenkes) [reply packages/telemetry/test/detectors/circleCiDetectorSync.spec.ts:34]: this is the reversing, we cache the values, delete them, then reset them to the cache

File: packages/telemetry/test/detectors/circleCiDetectorSync.spec.ts (hunk 5 above)
Comment (mjhenkes) [reply packages/telemetry/test/detectors/circleCiDetectorSync.spec.ts]: nope, not doing anything

File: packages/telemetry/test/detectors/circleCiDetectorSync.spec.ts (hunk 6 above)
Comment (AtofStryker) [reply packages/telemetry/test/detectors/circleCiDetectorSync.spec.ts:34]: Ah that makes a lot of sense now 😅

**Review by AtofStryker — APPROVED** [review]

**Pull request merged**